src/RolloutScorer/RolloutScorer/RolloutScorer.cs

```diff
@@ -49,85 +50,93 @@ public async Task InitAsync()
-                                    if (!previousAttemptTimelines.ContainsKey(attempt.TimelineId))
-                                    {
-                                        previousAttemptTimelines.Add(attempt.TimelineId,
-                                            (await GetAzdoApiResponseAsync($"{build.BuildSummary.TimelineLink}/{attempt.TimelineId}")).ToObject<BuildTimeline>());
-                                    }
-
-                                    if (previousAttemptTimelines[attempt.TimelineId] != null)
-                                    {
-                                        stages.Add(previousAttemptTimelines[attempt.TimelineId].Records
-                                            .Where(t => (t.Type == "Checkpoint.Approval" || t.Type == "Stage") && t.Name == record.Name).First());
-                                    }
+                                    previousAttemptTimelines.Add(attempt.TimelineId,
+                                        (await GetAzdoApiResponseAsync($"{buildBreakdown.BuildSummary.TimelineLink}/{attempt.TimelineId}")).ToObject<BuildTimeline>());
+                                }
+
+                                if (previousAttemptTimelines[attempt.TimelineId] != null)
+                                {
+                                    buildBreakdown.BuildSummary.Stages.Add(previousAttemptTimelines[attempt.TimelineId].Records
+                                        .Where(t => (t.Type == "Checkpoint.Approval" || t.Type == "Stage") && t.Name == record.Name).First());
                                 }
                             }
                         }
                     }
-                    duration = GetPipelineDurationFromStages(stages);
+                }
 
-                    if (stages.Any(s => s.Name.StartsWith("deploy", StringComparison.InvariantCultureIgnoreCase) && !string.IsNullOrEmpty(s.EndTime)))
-                    {
-                        build.BuildSummary.DeploymentReached = true;
-                    }
+                if (buildBreakdown.BuildSummary.Stages.Any(s => s.Name.StartsWith("deploy", StringComparison.InvariantCultureIgnoreCase) && !string.IsNullOrEmpty(s.EndTime)))
+                {
+                    buildBreakdown.BuildSummary.DeploymentReached = true;
                 }
+            }
+        }
+
+        /// <summary>
+        /// Calculates the "Time to Rollout" portion of the scorecard
+        /// </summary>
+        /// <returns>A timespan representing the total time to rollout and a list of scorecard build breakdowns for each rollout</returns>
+        public TimeSpan CalculateTimeToRollout()
+        {
+            List<TimeSpan> rolloutBuildTimes = new List<TimeSpan>();
+
+            // Loop over all the builds in the returned content and calculate start and end times
+            foreach (ScorecardBuildBreakdown build in BuildBreakdowns)
+            {
+                TimeSpan duration = TimeSpan.Zero;
```

Comment:
Why not just combine this line with line 139... why assign it to TimeSpan.Zero only to, on very next line, assign it to something else.

At the very least, there doesn't need to be a value here (since it's conditionally assigned on the next line)

Reply:
You can always set `GetPipelineDurationFromStages` to return TimeSpan.Zero in case of some unexpected calculation.

Reply:
I agree; this can be refactored better. Thanks!